### PR TITLE
Signals for properties

### PR DESCRIPTION
**Pull request by martin-ejdestig**

### CODE REVIEW

**Comment by martin-ejdestig** [comment]
There is a trivial conflict with #60 .

File: codegen_glibmm/templates/proxy.h.templ [hunk 1]
```diff
@@ -94,21 +96,25 @@ public:
...
-                       const Glib::VariantContainerBase &parameters);
-
 protected:
     Glib::RefPtr<Gio::DBus::Proxy> m_proxy;
 
 private:
     {{ interface.cpp_class_name }}(Glib::RefPtr<Gio::DBus::Proxy> proxy): Glib::ObjectBase() {
         this->m_proxy = proxy;
         this->m_proxy->signal_signal().connect(sigc::mem_fun(this, &{{ interface.cpp_class_name }}::handle_signal));
+        this->m_proxy->signal_properties_changed().connect(sigc::mem_fun(this, &{{ interface.cpp_class_name }}::handle_properties_changed));
```
Comment (mardy) [inline codegen_glibmm/templates/proxy.h.templ]: Wrap this line after the first `.`

File: codegen_glibmm/templates/proxy.h.templ [hunk 2]
```diff
@@ -83,6 +84,7 @@ public:
     void {{ prop.name }}_set({{ prop.cpptype_in }}, const Gio::SlotAsyncReady &);
     void {{ prop.name }}_set_finish(const Glib::RefPtr<Gio::AsyncResult> &);
     {% endif %}
+    sigc::signal<void> {{ prop.name }}_changed;
```
Comment (mardy) [inline codegen_glibmm/templates/proxy.h.templ]: I would prefer if this was a function: this in the future could allow us to refactor the way that signals work, for example avoid instantiating them until someone is actually interested in them.

File: codegen_glibmm/templates/proxy.h.templ (hunk 1 above)
Comment (mardy) [reply codegen_glibmm/templates/proxy.h.templ]: I would be happy to see this constructor moved to the cpp file; you don't have to do it, but, if you have some spare time, that would be great (but if you do it, do it in a separate commit) :-)

Comment (martin-ejdestig) [reply codegen_glibmm/templates/proxy.h.templ]: Done and done. (Thought about moving constructor before sending pr but held off due to conflict with #60.)

File: codegen_glibmm/templates/proxy.h.templ (hunk 2 above)
Comment (martin-ejdestig) [reply codegen_glibmm/templates/proxy.h.templ]: Yes, good idea. Especially if there are as many properties as in the many-types example.

**Comment by martin-ejdestig** [comment]
@mardy Ping. Would be nice to get this merged.

File: codegen_glibmm/templates/proxy.h.templ [hunk 3]
```diff
@@ -85,6 +86,10 @@ public:
     void {{ prop.name }}_set({{ prop.cpptype_in }}, const Gio::SlotAsyncReady &);
     void {{ prop.name }}_set_finish(const Glib::RefPtr<Gio::AsyncResult> &);
     {% endif %}
+    sigc::signal<void> &{{ prop.name }}_changed()
+    {
```
Comment (mardy) [inline codegen_glibmm/templates/proxy.h.templ]: move the `{` to the previous line

Comment (martin-ejdestig) [reply codegen_glibmm/templates/proxy.h.templ]: Done.

(A bit confused about what the style is. { on same line for short "getters" in headers and { on new line in .cpp ?)

Comment (mardy) [reply codegen_glibmm/templates/proxy.h.templ]: Yes, in the same line when the function is inline, on the next one otherwise.

**Pull request merged**